draft-ietf-protected-audience-key-value-services.md

```diff
@@ -320,28 +320,30 @@ Each key group is expected to have exactly one tag from the following list:
 
 ### Generating a Request {#request-generate}
 
-This section describes how the client MAY form and serialize request messages in order to fetch values from the Trusted Key Value server. 
+This section describes how the client MAY form and serialize request messages in order to fetch values from the Trusted Key Value server.
 
-This algorithm takes as input an [HPKE] `public key` and its associated `key id`, a `metadata` map for global configuration, where both keys and values are strings, and a list of `signals`, each of which is a map, from a list of strings to a list of strings.
+This algorithm takes as input an [HPKE] `public key` and its associated `key id`, a `metadata` map for global configuration, where both keys and values are strings, and a `compression groups` list, each of which is a map, from key `"partitions"` to value `partitions` as a list of maps.
 
 The output is an [HPKE] ciphertext encrypted `request` and a context `request context`.
 
 1. Let `request map` be an empty map.
-1. Let `compression group id` be 0.
 1. Let `partitions` be an empty array.
-1. For each `signal` in `signals`:
-    1. Let `partition` be an empty map.
-    1. Set `partition["id"]` to 0.
-    1. Set `partition["compressionGroupId"]` to `compression group id`.
-    1. Let `compression group id` increase by 1.
+1. For each `group` in `compression groups`:
+  1. For each `partition` in `compression groups["partitions"]:
```

Comment:
You're looping over `compression groups` and then over a field in  `compression groups`. It seems like that would give you all combinations of both, which is probably not what you want.